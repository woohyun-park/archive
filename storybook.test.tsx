import initStoryshots, { snapshotWithOptions } from "@storybook/addon-storyshots";

/* TODO: 현재 Textarea 컴포넌트가 react-textarea-autosize를 사용하는데,
 react-textarea-autosize가 내부적으로 forwardRef를 사용하는데,
 forwardRef는 element가 아니라서 테스트가 되지 않는다.

 임시적으로 document.createElement()를 통해서 패스처리를 해놓았으며,
 해당 오류를 우회할 수 있는 방법을 찾아봐야할듯하다. 
*/

initStoryshots({
  test: snapshotWithOptions((story: any) => ({
    createNodeMock: (element: Element) => {
      if (story.componentId === "textarea") {
        return document.createElement("textarea");
      }
      return element;
    },
  })),
});
