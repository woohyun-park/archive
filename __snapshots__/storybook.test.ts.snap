// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Button Default 1`] = `
<button
  className="flex items-center justify-center h-10 px-2 py-2 text-base text-center text-white transition duration-150 ease-in-out bg-black rounded-lg hover:bg-black-f focus:bg-black-f focus:outline-none focus:ring-0 active:bg-black-f min-w-fit"
  type="button"
>
  Button
</button>
`;

exports[`Storyshots Button Full 1`] = `
<button
  className="flex items-center justify-center h-10 px-2 py-2 text-base text-center text-white transition duration-150 ease-in-out bg-black rounded-lg hover:bg-black-f focus:bg-black-f focus:outline-none focus:ring-0 active:bg-black-f min-w-fit w-full"
  type="button"
>
  Button
</button>
`;

exports[`Storyshots Button Inactive 1`] = `
<button
  className="flex items-center justify-center h-10 px-2 py-2 text-base text-center transition duration-150 ease-in-out rounded-lg focus:outline-none focus:ring-0 min-w-fit text-black bg-gray-3 hover:bg-gray-3f focus:bg-gray-3f active:bg-gray-3f"
  type="button"
>
  Button
</button>
`;

exports[`Storyshots Button Small 1`] = `
<button
  className="flex items-center justify-center py-2 text-center text-white transition duration-150 ease-in-out bg-black rounded-lg hover:bg-black-f focus:bg-black-f focus:outline-none focus:ring-0 active:bg-black-f min-w-fit h-8 px-4 text-sm"
  type="button"
>
  Button
</button>
`;
